cmd/expired-authz-purger/main.go

```diff
@@ -71,33 +91,30 @@ func (p *expiredAuthzPurger) purgeAuthzs(purgeBefore time.Time, yes bool) (int64
 		}
 	}
 
-	rowsAffected := int64(0)
-	for {
-		result, err := p.db.Exec(`
-			DELETE FROM pendingAuthorizations
-			WHERE expires <= ?
-			LIMIT ?
-			`,
-			purgeBefore,
-			p.batchSize,
-		)
+	for _, id := range ids {
+		// Delete challenges + authorization
+		_, err := p.db.Exec("DELETE FROM challenges WHERE authorizationID = ?", id)
 		if err != nil {
-			return rowsAffected, err
+			return err
 		}
-		rows, err := result.RowsAffected()
+		_, err = p.db.Exec(fmt.Sprintf("DELETE FROM %s WHERE id = ?", table), id)
 		if err != nil {
-			return rowsAffected, err
+			return err
 		}
+	}
 
-		p.stats.Inc("PendingAuthzDeleted", rows)
-		rowsAffected += rows
-		p.log.Info(fmt.Sprintf("Progress: Deleted %d (%d total) expired pending authorizations", rows, rowsAffected))
+	p.log.Info(fmt.Sprintf("Deleted a total of %d expired authorizations from %s", len(ids), table))
+	return nil
+}
 
-		if rows < p.batchSize {
-			p.log.Info(fmt.Sprintf("Deleted a total of %d expired pending authorizations", rowsAffected))
-			return rowsAffected, nil
+func (p *expiredAuthzPurger) purgeAuthzs(purgeBefore time.Time, yes bool) error {
+	for _, table := range []string{"pendingAuthorizations", "authz"} {
```

Comment:
I think it makes sense to do this as a follow up, it would also make life easier if/when we depreciate the pending/finalized authorization table split.